src/Faker/Name.cs

```diff
@@ -1,56 +1,60 @@
-﻿using System;
+﻿using Faker.Extensions;
+using System;
 using System.Collections.Generic;
 using System.Linq;
-using Faker.Extensions;
 
 namespace Faker
 {
 	/// <summary>
-	///     Enumeration of which Name format to use in the <see cref="Name.FullName(NameFormats)" /> method.
+	///   Enumeration of which Name format to use in the <see cref="Name.FullName(NameFormats)" /> method.
 	/// </summary>
-	/// <include file='Docs/NameFormatsExample.xml' path='example' />
+	/// <include file="Docs/NameFormatsExample.xml" path="example" />
 	[Flags]
 	public enum NameFormats : byte
 	{
 		/// <summary>
-		///     The standard (First and Last name only).
+		///   The standard (First and Last name only).
 		/// </summary>
 		Standard = 1 << 0,
 
 		/// <summary>
-		///     First and Last name with a Prefix.
+		///   First and Last name with a Prefix.
 		/// </summary>
 		WithPrefix = 1 << 1,
 
 		/// <summary>
-		///     First and Last name with a Suffix.
+		///   First and Last name with a Suffix.
 		/// </summary>
 		WithSuffix = 1 << 2,
 
 		/// <summary>
-		///     First Last name with both a Prefix and a Suffix.
+		///   First Last name with both a Prefix and a Suffix.
 		/// </summary>
 		WithPrefixAndSuffix = WithPrefix | WithSuffix
 	}
 
 	/// <summary>
-	///     A collection of Personal name related resources.
+	///   A collection of Personal name related resources.
 	/// </summary>
-	/// <include file='Docs/CustomRemarks.xml' path='Comments/SatelliteResource/*' />
+	/// <include file="Docs/CustomRemarks.xml" path="Comments/SatelliteResource/*" />
 	/// <threadsafety static="true" />
 	public static class Name
 	{
 		/// <summary>
-		///     Creates a random first name.
+		///   Creates a random first name.
 		/// </summary>
 		/// <returns>The randomly created first name.</returns>
 		public static string First()
 		{
-			return Resources.Name.First.Split(Config.SEPARATOR).Random().Trim();
+			var cultureInfo = System.Globalization.CultureInfo.CurrentCulture.TwoLetterISOLanguageName;
+			if (!firstArrayCacheDictionary.ContainsKey(cultureInfo))
+				firstArrayCacheDictionary[cultureInfo] = Resources.Name.First.Split(Config.SEPARATOR).Select(s => s.Trim()).ToArray();
+
```

Comment:
These changes makes this not Thread safe,
if two threads access the method at the exact same time who will set the value,
the first one, second one, both or throw an error.
Must be inside a lock although that will take a performance hit.